packages/replay-next/src/suspense/SyntaxParsingCache.ts

```diff
@@ -1,407 +1,157 @@
-  const parser = incrementalParser(fileName);
-  if (parser === null) {
-    return null;
-  }
-
-  parser.parseChunk(code, true);
-
-  return parser.parsedTokensByLine;
+export interface PrepareRequest {
+  type: "prepare";
+  sourceId: SourceId;
+  fileName: string | null;
+  contentType?: ContentType | null;
 }
 
-function identity(any: any) {
-  return any;
+export interface ParseRequest {
+  type: "parse";
+  sourceId: SourceId;
+  chunk: string;
+  isLastChunk: boolean;
 }
 
-function contentTypeToLanguage(contentType: ContentType): LRLanguage {
-  switch (contentType) {
-    case "text/html":
-      return htmlLanguage;
-    case "text/javascript":
-      return javascriptLanguage;
-    default:
-      console.error(`Unknown content-type: "${contentType}"`);
-      return javascriptLanguage;
-  }
+export interface ParseResponse {
+  sourceId: SourceId;
+  parsedTokens: ParsedToken[][];
+  totalParsedLength: number;
 }
 
-function urlToLanguage(fileName: string): LRLanguage {
-  const extension = fileName.split(".").pop()!.split("?").shift()!;
-  switch (extension) {
-    case "js":
-      return javascriptLanguage;
-    case "jsx":
-      return jsxLanguage;
-    case "ts":
-      return typescriptLanguage;
-    case "tsx":
-      return tsxLanguage;
-    case "json":
-      return jsonLanguage;
-    case "html":
-      return htmlLanguage;
-    default:
-      console.error(`Unknown file extension: "${extension}"`);
-      return javascriptLanguage;
+let worker: Worker | undefined;
+function postWorkerMessage(message: PrepareRequest | ParseRequest) {
+  if (!worker) {
+    // @ts-ignore
+    worker = new Worker(new URL("./SyntaxParsingCacheWorker", import.meta.url));
+    worker.onmessage = handleParseResponse;
   }
+  worker.postMessage(message);
 }
 
-function processSection(
-  currentLineState: {
-    parsedTokens: ParsedToken[];
-    rawString: string;
-  },
-  parsedTokens: Array<ParsedToken[]>,
-  section: string,
-  className: string
-) {
-  const tokenTypes = className ? classNameToTokenTypes(className) : null; // Remove "tok-" prefix;
-
-  let index = 0;
-  let nextIndex = section.indexOf("\n");
-
-  while (true) {
-    const substring =
-      nextIndex >= 0 ? section.substring(index, nextIndex) : section.substring(index);
-
-    const token: ParsedToken = {
-      columnIndex: currentLineState.rawString.length,
-      types: tokenTypes,
-      value: substring,
-    };
-
-    currentLineState.parsedTokens.push(token);
-    currentLineState.rawString += substring;
-
-    if (nextIndex === -1) {
-      break;
-    }
-
-    if (nextIndex >= 0) {
-      parsedTokens.push(currentLineState.parsedTokens);
-
-      currentLineState.parsedTokens = [];
-      currentLineState.rawString = "";
-    }
-
-    index = nextIndex + 1;
-    nextIndex = section.indexOf("\n", index);
+function handleParseResponse({ data }: MessageEvent<ParseResponse>) {
+  const state = internalParserStates.get(data.sourceId);
```

Comment:
So this `data` message is _guaranteed_ to always break evenly at the line level? Even for sources with _really long lines_?

Reply:
What do you mean by "break evenly at the line level"?

Reply:
I mean that the parsing will have to always finish at the line boundary

Reply:
OK, yeah, `data` will only contain completed lines: the worker removes the last line of a chunk if it isn't the last chunk, because that line will usually be incomplete. When the next chunk comes in, it doesn't start highlighting at the beginning of that chunk but at the beginning of the first line that hasn't been highlighted yet.

Reply:
Great! Thanks. Just wanted to confirm that behavior was still in place.